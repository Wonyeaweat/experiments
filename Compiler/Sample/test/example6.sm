program example6;
var  A,B,C,D:integer;
begin
while A < C and B < D do 
    if A = 1 then
       C := C + 1
    else while A <= D do
       A := A + 2
end.