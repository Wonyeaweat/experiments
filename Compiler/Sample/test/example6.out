( 0) (program, example6, - , - )
( 1) (j<, A , C , 3 )
( 2) (j , - , - , 16)
( 3) (j<, B , D , 5 )
( 4) (j , - , - , 16)
( 5) (j=, A , 1 , 7 )
( 6) (j , - , - , 10)
( 7) (+ , C , 1 , T1)
( 8) (:=, T1, - , C )
( 9) (j , - , - , 1 )
(10) (j<=, A , D , 12)
(11) (j , - , - , 1 )
(12) (+ , A , 2 , T2)
(13) (:=, T2, - , A )
(14) (j , - , - , 10)
(15) (j , - , - , 1 )
(16) (sys, - , - , - )
